github-for-jira.sd.yml

```diff
@@ -149,6 +149,16 @@ resources:
         # Name of the RDS resource from above
         resource: rds
 
+  - name: deployment-history
+    type: dynamo-db
+    attributes: &table-attributes
+      HashKeyName: Id
```

Comment:
nit: could it be something more descriptive?

Reply:
Yeah, I was thinking of some more explicit naming, e.g. InstallationIdEnvRepoId. I don't think you can easily migrate the records from one logic to another, so I'd not have bothered about flexibility. Also, the name of the range key you are using is quite descriptive 😛

Reply:
`migrate the records from one logic to another` <--- less about this (data migration), more about ask micros/aws to change the partition key from `InstallationIdEnvRepoId` to `AnotherIdEnvRepoId`? Is it possible?

Reply:
> Yeah, I was thinking of some more explicit naming, e.g. InstallationIdEnvRepoId. I don't think you can easily migrate the records from one logic to another, so I'd not have bothered about flexibility. Also, the name of the range key you are using is quite descriptive 😛

I almost change the `StatusCreatedAt` to `CreatedAt`.

Reply:
>  more about ask micros/aws to change the partition key from InstallationIdEnvRepoId to AnotherIdEnvRepoId

I reckon for a migration like that you'll need to create a new table anyway, wdyt?

Reply:
I agree, and that's why I make the ID column generic. But also agree no easy way to migrate data even I can getaway with the generic ID col. 

In short, to change the semantic, nothing easy.

Reply:
Okay I've tested, there's no easy way to change the partition key schema on existing table.
After another thought, I still prefer to have a generic Id column, so that when we change our mind in the future, we can:
1. Save the db with new key value
2. Ignore the existing data as a) they will not conflicts with new value (hash + new semantic) and b) they will expired by themselves. 
3. We don't have to deploy a new table with new name + use ff to switch over + cli on prod ( + stg/dev) to remove unused table. 
Too much effort involved to change to use another table should we change our mind to use another value for key.